app/src/lib/git/credential.ts

```diff
@@ -0,0 +1,52 @@
+import { GitProcess } from 'dugite'
+
+function parseCredential(output: string) {
+  const kv = new Map<string, string>()
+  for (const [, k, v] of output.matchAll(/^(.*?)=(.*)$/gm)) {
+    kv.set(k, v)
+  }
+  return kv
+}
+
+function formatCredential(credential: Map<string, string>) {
+  return [...credential].map(([k, v]) => `${k}=${v}\n`).join('')
+}
+
+const execCredential = (args: string[], path: string, stdin?: string) =>
+  // Can't use git() as that will call withTrampolineEnv which calls this method
+  GitProcess.exec(
+    [
+      ...['-c', 'credential.helper='],
+      ...['-c', `credential.helper=manager`],
+      'credential',
+      ...args,
+    ],
+    path,
+    {
+      stdin,
+      env: {
+        GIT_TERMINAL_PROMPT: '0',
+        GIT_ASKPASS: '',
+        TERM: 'dumb',
+      },
+    }
+  ).then(r => {
+    if (r.exitCode !== 0) {
+      // TODO: can we handle this error like we handle other errors in git()?
+      throw new Error(r.stderr)
+    }
+    return parseCredential(r.stdout)
+  })
```

Comment:
Desktop ships with mingit on Windows, which includes GCM so `-c credential.helper=manager` should always be successful.

What happens on macOS and Linux when the user has not installed GCM? I'd imagine a generic error from Git as:

```shell
% git -c credential.helper= -c credential.helper=manager credential fill
git: 'credential-manager' is not a git command. See 'git --help'.
```

How does this surface to the user?

Reply:
Great question. GitHub Desktop ships with its own Git environment (https://github.com/desktop/dugite) where we bundle GCM for macOS and Linux (https://github.com/desktop/dugite-native/pull/510).